From 9fbca4af083e29edacb620537466d1cd20914c63 Mon Sep 17 00:00:00 2001
From: Alec Leamas <alec@tests.notat.diaspora.com>
Date: Tue, 30 Oct 2012 20:38:19 +0100
Subject: [PATCH 3/3] Don't configure out-of-tree components.

Avoid creation of Makefile in components where external ones are
instead (lpc10, libgsm).
---
 configure.ac | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9117695..bfc804f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -602,23 +602,23 @@ AC_SUBST(PLAYRECLINKS)
 
 AM_CONDITIONAL(STATIC_LIBSOX_ONLY, test "$enable_shared" = "no" -a "$enable_static" = "yes")
 
-dnl Generate output files.
-AC_CONFIG_FILES(Makefile src/Makefile libgsm/Makefile lpc10/Makefile msvc9/Makefile msvc10/Makefile sox.pc)
-AC_OUTPUT
-
 if test "$using_gsm" != "no"; then
   if test "$found_libgsm" = "yes"; then
     gsm_option="(external)"
+    gsm_makefile=""
   else
     gsm_option="(in-tree)"
+    gsm_makefile="libgsm/Makefile"
   fi
 fi
 
 if test "$using_lpc10" != "no"; then
   if test "$found_liblpc10" = "yes"; then
     lpc10_option="(external)"
+    lpc10_makefile=""
   else
     lpc10_option="(in-tree)"
+    lpc10_makefile="lpc10/Makefile"
   fi
 fi
 
@@ -628,6 +628,11 @@ else
   pkgconfig_option="$with_pkgconfigdir"
 fi
 
+dnl Generate output files.
+AC_CONFIG_FILES(Makefile src/Makefile $libgsm_makefile $lpc10_makefile msvc9/Makefile msvc10/Makefile sox.pc)
+
+AC_OUTPUT
+
 # Report configuration.
 echo
 echo "BUILD OPTIONS"
-- 
1.7.11.4

